modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2Tree.java

```diff
@@ -190,7 +209,7 @@ private boolean inlineObjectSupported(MetaPageInfo metaInfo, List<InlineIndexHel
             try {
                 if (H2TreeInlineObjectDetector.objectMayBeInlined(inlineSize, inlineIdxs)) {
```

Comment:
no, for that case we can't have a null values - here present only types of columns

Reply:
I meant the code
```
public static boolean objectMayBeInlined(int inlineSize, List<InlineIndexHelper> inlineHelpers) {
    int remainSize = inlineSize;

    for (InlineIndexHelper ih : inlineHelpers) {
        if (ih.type() == Value.JAVA_OBJECT)
            break;

        remainSize -= ih.size() > 0 ? 1 + ih.size() : 4;
    }

    return remainSize >= 4;
}
```
It calculates remainSize not taking into account that `null` might be inlined for some column. So, we may decide here that there is not enough space for inlining JAVA_OBJECT but actually it can be inlined in a tree.

Reply:
Was the comment above addressed?

Reply:
thanks for the memo. Will amend 4 to 1 for varLen types